en/developer/customization/articles/50-liferay-forms/03-creating-form-storage-adapters.markdown

```diff
@@ -6,27 +6,247 @@ header-id: creating-a-form-storage-adapter
 
 [TOC levels=1-4]
 
-There's only one class to create when implementing a Form Storage Adapter, and
-it extends the base `StorageAdapter` implementation.
+To create a storage adapter, implement the `DDMStorageAdapter` interface. The
+[`DDMJSONStorageAdapter`](https://github.com/liferay/liferay-portal/blob/7.2.x/modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/storage/DDMJSONStorageAdapter.java)
+is @product@'s default implementation.
+
+## Step 1: Declare the Implementation as a Component
+
+In the component declaration, set the `ddm.storage.adapter.type` property so
+that the service registry can retrieve your implementation.
 
 ```java
-@Component(service = StorageAdapter.class)
-public class FileSystemStorageAdapter extends BaseStorageAdapter {
+@Component(
+	immediate = true, property = "ddm.storage.adapter.type=file",
+	service = DDMStorageAdapter.class
+)
+public class DDMFileSystemStorageAdapter implements DDMStorageAdapter {
 ```
 
+<!-->
 The only method without a base implementation in the abstract class is
 `getStorageType`. For file system storage, it can return `"File System"`.
 
 ```java
 @Override
 public String getStorageType() {
-    return "File System";
+return "File System";
+}
+```
+</!-->
```

Comment:
All this block should be removed, once it's related to the `BaseStorageAdapter` class.

Reply:
Thanks Renato, this article isn't ready for review yet.

Reply:
Sorry, Russ. Once I needed to read it all, I was adding comments to everything I thought that requires a check from your side. 😁